apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2411,3 +2411,38 @@ func (u *UniterAPI) SetContainerSpec(args params.SetContainerSpecParams) (params
 	}
 	return results, nil
 }
+
+// GoalStates returns the list with charm peers and related units information.
+func (u *UniterAPI) GoalStates(args params.Entities) (params.GoalStateResults, error) {
+	result := params.GoalStateResults{
+		Results: make([]params.GoalStateResult, len(args.Entities)),
+	}
+	canAccess, err := u.accessUnit()
+	if err != nil {
+		return params.GoalStateResults{}, err
+	}
+	for i, entity := range args.Entities {
+		tag, err := names.ParseUnitTag(entity.Tag)
+		if err != nil {
+			result.Results[i].Error = common.ServerError(common.ErrPerm)
+			continue
+		}
+		if !canAccess(tag) {
+			result.Results[i].Error = common.ServerError(common.ErrPerm)
+			continue
+		}
+		unit, err := u.getUnit(tag)
+		if err != nil {
+			result.Results[i].Error = common.ServerError(err)
+			continue
+		}
+		machineId, err := unit.AssignedMachineId()
```

Comment:
remove this assigned machine stuff and just use return the yaml of an empty goal state struct or something